src/test/optimizer/strategy/predicate_pushdown_rule_test.cpp

```diff
@@ -138,4 +139,98 @@ TEST_F(PredicatePushdownRuleTest, ComplexBlockingPredicatesPushdownTest) {
   EXPECT_EQ(reordered->left_input()->right_input()->left_input(), _table_a);
 }
 
+TEST_F(PredicatePushdownRuleTest, AllowedValuePredicatePushdownThroughProjectionTest) {
```

Comment:
Goes for all the new tests
* Code duplication: `lqp_c, int_float_node_a, ...` can you init all these in `class PredicatePushdownRuleTest` to cleanly separate setup code from actual test code
* I know you just copied and adapted these, but they would really gain readability if they would follow the "new" pattern/formatting of LQP testing: Build an `expected_lqp`, let hyrise generate an `actual_lqp`, check them with `EXPECT_LQP_EQ`.

```c++
TEST_F(SQLTranslatorTest, WhereSimple) {
  const auto actual_lqp = compile_query("SELECT a FROM int_float WHERE a < 200;");

  // clang-format off
  const auto expected_lqp =
  ProjectionNode::make(expression_vector(int_float_a),
    PredicateNode::make(less_than_(int_float_a, 200), stored_table_node_int_float));
  // clang-format on

  EXPECT_LQP_EQ(actual_lqp, expected_lqp);
}
```

Reply:
IMO, when implementing optimizer rules you have to make sure the rule doesn't break the LQP. Just testing the presence of nodes can easily miss bugs.

Second point: If you test the entire LQP, then the reader of the code can be sure the test actually tests everything the rule might or might not do to the LQP. If you just test a selected number of node types, the reader really has to study your test to figure out whether it might miss anything. Which the reader won't do, cause he lazy. Enter error proneness.

Reply:
Ok, but this should be part of a bigger cleanup of the rule tests. Doing it one way in these new tests and differently a couple of lines above doesn't seem right, either.